src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -139,28 +139,28 @@ private static bool TryUseSubstring(ReadOnlySpan<string> uniqueStrings, bool ign
                     // If the substring isn't ASCII, bail out to return the results.
                     if (!IsAllAscii(substring))
                     {
-                        allAsciiIfIgnoreCase = false;
-                        canSwitchIgnoreCaseToCaseSensitive = false;
+                        allAsciiIfIgnoreCaseForHash = false;
```

Comment:
Would it be possible to also undo the variable renamings to trim down the diff to the actual behavioral change?